konduit-serving-pipeline/src/test/java/ai/konduit/serving/pipeline/impl/data/JsonCoverageTest.java

```diff
@@ -0,0 +1,251 @@
+                .returnProb(true)
+                .allProbabilities(true)
+                .labelName("label")
+                .topN(2)
+                .indexName("index")
+                .probName("prob"));
+    }
+
+
+    @Test
+    public void testSimpleLoopTriggerT() {
+        testConfigSerDe(new SimpleLoopTrigger(500L));
+    }
+
+    @Test
+    public void testTimeLoopTrigger() {
+        testConfigSerDe(new TimeLoopTrigger(1, TimeUnit.MINUTES));
+    }
+
+
+    @Test
+    public void testTimeStepTest() {
+        testConfigSerDe(new TimeStep());
+    }
+
+
+    @Test
+    public void testCountStepTest() {
+        testConfigSerDe(new CountStep(1));
+    }
+
+    @Test
+    public void testSwitchFn() {
+        testConfigSerDe(new TestSwitchFn().branch(1));
+    }
+
+    @Test
+    public void testDataStringSwitchFn() {
+        Map<String, Integer> hashMap = new HashMap<String, Integer>();
+        hashMap.put("a", 0);
+        testConfigSerDe(new DataStringSwitchFn(1, "foo", hashMap));
+    }
+
+    @Test
+    public void testDataIntSwitchFn() {
+        testConfigSerDe(new DataIntSwitchFn(1, "foo"));
+    }
+
+    @Test
+    public void testSwitchStep() {
+        testConfigSerDe(new SwitchStep("foo", null));
+    }
+
+    @Test
+    public void testGraphPipeline() {
+        Map<String, GraphStep> steps = new HashMap<String, GraphStep>();
+        testConfigSerDe(new GraphPipeline(steps, "foo", "myGraphPipeline"));
+    }
+
+    @Test
+    public void testAsyncPipeline() {
+        Map<String, GraphStep> steps = new HashMap<String, GraphStep>();
+        testConfigSerDe(new ai.konduit.serving.pipeline.impl.pipeline.AsyncPipeline(new GraphPipeline(steps, "foo", "myGraphPipeline"), new SimpleLoopTrigger()) {
+        });
+    }
+
+
+    @Test
+    public void testSequencePipeline() {
+        testConfigSerDe(new SequencePipeline(new ArrayList<PipelineStep>(), "foo"));
+    }
+
+
+    @Test
+    public void testAnyStep() {
+        testConfigSerDe(new AnyStep(new GraphBuilder(), new ArrayList<String>(), "foo"));
+    }
+
+
+    @Test
+    public void testMergeStep() {
+        testConfigSerDe(new MergeStep(new GraphBuilder(), new ArrayList<String>(), "foo"));
+    }
+
+
+    @Test
+    public void testPipelineGraphStep() {
+        testConfigSerDe(new PipelineGraphStep(new GraphBuilder(), new BoundingBoxToPointStep(), "step", "name"));
+    }
+
+    @Test
+    public void testSwitchOutput() {
+        testConfigSerDe(new SwitchOutput(new GraphBuilder(), "foo", "bar", 2));
+    }
+
+
+    // Failed tests
+
+    // fails with https://gist.github.com/atuzhykov/666a99c9dd540a57437750ac31114528
+    @Test
+    @Ignore
+    public void testInput() {
+        testConfigSerDe(new Input(new GraphBuilder().id("foo")));
+    }
+
+    // fails with https://gist.github.com/atuzhykov/fc87f4064ad00300da34403857d15d55
+    @Test
+    @Ignore
+    public void testPipelineProfilerTest() {
+        testConfigSerDe(new PipelineProfilerTest());
```

Comment:
This is a test class, we don't need to test any of the test classes for JSON/YAML.
Let's add a method in BaseJsonCoverageTest like this:
```
public Set<Class<?>> ignores(){
    return Collections.emptySet();
}
```

For modules like this, we override that method in JsonCoverageTest and return the test classes.

Then in the for loop in `@Before` (i.e., in `for (Class<?> c : subTypes) {`) ignore if the class is in the set returned by ignores() method.

PipelineProfilerTest, FunctionStep, CallbackStep, are all like this


As for Input GraphStep (previous test) - that should never appear in the JSON (it's just for building) so let's ignore that one too.